x/auth/client/testutil/helpers.go

```diff
@@ -1,60 +1,116 @@
 }
 
-// TxValidateSignatures is simcli tx validate-signatures
-func TxValidateSignatures(f *cli.Fixtures, fileName string, flags ...string) (bool, string, string) {
-	cmd := fmt.Sprintf("%s tx validate-signatures %v --keyring-backend=test %v", f.SimdBinary,
-		f.Flags(), fileName)
+func TxValidateSignaturesExec(clientCtx client.Context, filename string) ([]byte, error) {
+	args := []string{
+		fmt.Sprintf("--%s=%s", flags.FlagKeyringBackend, keyring.BackendTest),
+		fmt.Sprintf("--%s=%s", flags.FlagChainID, clientCtx.ChainID),
+		filename,
+	}
 
-	return cli.ExecuteWriteRetStdStreams(f.T, cli.AddFlags(cmd, flags), clientkeys.DefaultKeyPass)
+	return callCliCmd(clientCtx, cli.GetValidateSignaturesCommand, args)
 }
 
-// TxMultisign is simcli tx multisign
-func TxMultisign(f *cli.Fixtures, fileName, name string, signaturesFiles []string,
-	flags ...string) (bool, string, string) {
+func TxMultiSignExec(clientCtx client.Context, from string, filename string, extraArgs ...string) ([]byte, error) {
+	args := []string{
+		fmt.Sprintf("--%s=%s", flags.FlagKeyringBackend, keyring.BackendTest),
+		fmt.Sprintf("--%s=%s", flags.FlagChainID, clientCtx.ChainID),
+		filename,
+		from,
+	}
 
-	cmd := fmt.Sprintf("%s tx multisign --keyring-backend=test %v %s %s %s", f.SimdBinary, f.Flags(),
-		fileName, name, strings.Join(signaturesFiles, " "),
-	)
-	return cli.ExecuteWriteRetStdStreams(f.T, cli.AddFlags(cmd, flags))
+	args = append(args, extraArgs...)
+
+	return callCliCmd(clientCtx, cli.GetMultiSignCommand, args)
 }
 
-func TxSignBatch(f *cli.Fixtures, signer, fileName string, flags ...string) (bool, string, string) {
-	cmd := fmt.Sprintf("%s tx sign-batch %v --keyring-backend=test --from=%s %v", f.SimdBinary, f.Flags(), signer, fileName)
+func TxSignBatchExec(clientCtx client.Context, from fmt.Stringer, filename string, extraArgs ...string) ([]byte, error) {
+	args := []string{
+		fmt.Sprintf("--%s=%s", flags.FlagKeyringBackend, keyring.BackendTest),
+		fmt.Sprintf("--from=%s", from.String()),
+		filename,
+	}
+
+	args = append(args, extraArgs...)
 
-	return cli.ExecuteWriteRetStdStreams(f.T, cli.AddFlags(cmd, flags), clientkeys.DefaultKeyPass)
+	return callCliCmd(clientCtx, cli.GetSignBatchCommand, args)
 }
 
-// TxDecode is simcli tx decode
-func TxDecode(f *cli.Fixtures, encodedTx string, flags ...string) (bool, string, string) {
-	cmd := fmt.Sprintf("%s tx decode %v %v", f.SimdBinary, f.Flags(), encodedTx)
-	return cli.ExecuteWriteRetStdStreams(f.T, cli.AddFlags(cmd, flags), clientkeys.DefaultKeyPass)
+func TxDecodeExec(clientCtx client.Context, encodedTx string, extraArgs ...string) ([]byte, error) {
+	args := []string{
+		fmt.Sprintf("--%s=%s", flags.FlagKeyringBackend, keyring.BackendTest),
+		encodedTx,
+	}
+
+	args = append(args, extraArgs...)
+
+	return callCliCmd(clientCtx, cli.GetDecodeCommand, args)
+}
+
+func callCliCmd(clientCtx client.Context, theCmd func() *cobra.Command, extraArgs []string) ([]byte, error) {
```

Comment:
I guess it is worth have an auxiliary function for this.

1. Rename it: `ExecTestCLICmd`
2. Why does it take a command function, and not just a `*cobra.Command` literal?
3. Move this to the `testutils` pkg.

Reply:
Check now, I have put it back to the testutil but now there is lots of problems of import cycle, so I put it into testutil/cli. If we want to change that and make it work into testutil I will open an issue because everytime I get new problems with this PR when merging master.

Reply:
Looks great! I approve! Just left one minor remark.